rmgpy/molecule/parserTest.py

```diff
@@ -48,6 +59,264 @@ def test_fromSMILES(self):
+
+
+        # Test CO2
+        adjlist = '''
+        1 O u0 p2 c0 {2,D}
+        2 C u0 p0 c0 {1,D} {3,D}
+        3 O u0 p2 c0 {2,D}
+        '''
+        smiles = 'O=C=O'
+        self.compare(adjlist, smiles)
+
+
+        # Test CO
+        adjlist = '''
+        1 C u0 p1 c-1 {2,T}
+        2 O u0 p1 c+1 {1,T}
+        '''
+        smiles = '[C-]#[O+]'
+        self.compare(adjlist, smiles)
+
+
+        # Test C2H4
+        adjlist = '''
+        1 C u0 p0 c0 {2,D} {3,S} {4,S}
+        2 C u0 p0 c0 {1,D} {5,S} {6,S}
+        3 H u0 p0 c0 {1,S}
+        4 H u0 p0 c0 {1,S}
+        5 H u0 p0 c0 {2,S}
+        6 H u0 p0 c0 {2,S}
+        '''
+        smiles = 'C=C'
+        self.compare(adjlist, smiles)
+
+
+        # Test O2
+        adjlist = '''
+        1 O u0 p2 c0 {2,D}
+        2 O u0 p2 c0 {1,D}
+        '''
+        smiles = 'O=O'
+        self.compare(adjlist, smiles)
+
+
+        # Test CH3
+        adjlist = '''
+        multiplicity 2
+        1 C u1 p0 c0 {2,S} {3,S} {4,S}
+        2 H u0 p0 c0 {1,S}
+        3 H u0 p0 c0 {1,S}
+        4 H u0 p0 c0 {1,S}
+        '''
+        smiles = '[CH3]'
+        self.compare(adjlist, smiles)
+
+
+        # Test HO
+        adjlist = '''
+        multiplicity 2
+        1 O u1 p2 c0 {2,S}
+        2 H u0 p0 c0 {1,S}
+        '''
+        smiles = '[OH]'
+        self.compare(adjlist, smiles)
+
+
+        # Test C2H5
+        adjlist = '''
+        multiplicity 2
+        1 C u0 p0 c0 {2,S} {5,S} {6,S} {7,S}
+        2 C u1 p0 c0 {1,S} {3,S} {4,S}
+        3 H u0 p0 c0 {2,S}
+        4 H u0 p0 c0 {2,S}
+        5 H u0 p0 c0 {1,S}
+        6 H u0 p0 c0 {1,S}
+        7 H u0 p0 c0 {1,S}
+        '''
+        smiles = 'C[CH2]'
+        self.compare(adjlist, smiles)
+
+
+        # Test O
+        adjlist = '''
+        multiplicity 3
+        1 O u2 p2 c0
+        '''
+        smiles = '[O]'
+        self.compare(adjlist, smiles)
+
+
+        # Test HO2
+        adjlist = '''
+        multiplicity 2
+        1 O u1 p2 c0 {2,S}
+        2 O u0 p2 c0 {1,S} {3,S}
+        3 H u0 p0 c0 {2,S}
+        '''
+        smiles = '[O]O'
+        self.compare(adjlist, smiles)
+
+
+        # Test CH, methylidyne.
+        # Wikipedia reports:
+        # The ground state is a doublet radical with one unpaired electron,
+        # and the first two excited states are a quartet radical with three
+        # unpaired electrons and a doublet radical with one unpaired electron.
+        # With the quartet radical only 71 kJ above the ground state, a sample
+        # of methylidyne exists as a mixture of electronic states even at
+        # room temperature, giving rise to complex reactions.
+        #
+        # Should we make the doublet? For now, this is a regression test,
+        # because we currently make the quartet.
+        adjlist = '''
+        multiplicity 4
+        1 C u3 p0 c0 {2,S}
+        2 H u0 p0 c0 {1,S}
+        '''
+        smiles = '[CH]'
+        self.compare(adjlist, smiles)
```

Comment:
since this PR fixes `[C]`, maybe we should fix this `[CH]` as well? I'll add one commit shortly.